MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc

```diff
@@ -53,6 +53,10 @@ volumeHandle::volumeHandle(const cms::DDFilteredView &fv, bool expand2Pi, bool d
     case DDSolidShape::ddtubs:
       buildTubs();
       break;
+    case DDSolidShape::ddpseudotrap: {
+      vector<double> d = solid.volume().volume().solid().dimensions();
```

Comment:
```suggestion
      vector<double> d = solid.parameters();
```

Reply:
Let's avoid internal DD4hep implementation details by using the `parameters` method.

Reply:
`solid.volume().volume().solid().dimensions() `is exactly the workaround I had to figure out to avoid the crash that otherwise happens when asking parameters() on a PseudoTrap.
For the records, I've debugged the crash to happen after this if evaluates to true:
https://github.com/cms-sw/cmssw/blob/master/DetectorDescription/DDCMS/src/DDFilteredView.cc#L452

in fact, for some reason beyond my understanding, currVol->GetShape()->IsA() turns out to be TGeoCompositeShape::Class() for a PseudoTrap. Then the code to handle a composite object is executed, and this leads to something unexpected. 
I verified that there would be no crash if that if() evaluated to false, but to me there is no way to add a protection without adding some specific condition for PseudoTraps, which I guess we want to avoid here.

So I figured out that it is a better idea of putting a small workaround in MF, which is in a very closed scope, than to hack DDFilteredView for a very specific use case.

Of course if you know how to fix DDFilteredView so that it gives dimensions() for PseudoTraps without crashing, that would be welcome.
I was con

Reply:
We have two issues here. Line 57 is vulnerable to any internal change to DD4hep. I can tell you from experience that DD4hep changes with some frequency, even in its external interface, so I would worry that some time years from now, the DD4hep internals change to break this line, and we will have all forgotten about the special pseudotrapezoids and will have a hard time figuring out how to fix the problem.

Reply:
The second issue is that a `DDFilteredView` method should never crash, regardless of what shape is involved. Yana and I need to fix `parameters` to make it more robust.
@ianna FYI `DDFilteredView::parameters` crashes with a pseudotrapezoid.

Reply:
This PR can go ahead while we figure out how to deal with these issues. Use the version of line 57 that you found works.

Reply:
@cvuosalo, I fully agree. This fix is intended to keep legacy maps working while we have no better solution; it is intended that it might need to be adapted if the underlying interface changes.
A fix to DDFilteredView::parameters() would be very useful for this purpose. I spent some time on it but could not find an acceptable fix myself. I can produce a simple recipe to reproduce the crash if that can help others to work on it.